src/site_upload/unzip_upload/unzip_upload.py

```diff
@@ -0,0 +1,60 @@
+"""Lambda for moving data to processing locations"""
+
+import datetime
+import logging
+import os
+import zipfile
+from io import BytesIO
+
+import boto3
+
+from shared import decorators, enums, functions
+
+log_level = os.environ.get("LAMBDA_LOG_LEVEL", "INFO")
+logger = logging.getLogger()
+logger.setLevel(log_level)
+
+
+def unzip_upload(s3_client, sns_client, s3_bucket_name: str, s3_key: str) -> None:
+    metadata = functions.parse_s3_key(s3_key)
+    buffer = BytesIO(s3_client.get_object(Bucket=s3_bucket_name, Key=s3_key)["Body"].read())
+    archive = zipfile.ZipFile(buffer)
+    files = archive.namelist()
+    files.remove("manifest.toml")
+    meta_files = [x for x in files if "__meta_" in x]
+    files = list(set(files) - set(meta_files) - set(["manifest.toml"]))
+    # The manifest will be used as a signal that the extract has finished,
+    # so we'll extract it last in all cases.
+    # TODO: decide on extract location for manifests
+    for file_list in [meta_files, files, ["manifest.toml"]]:
+        for file in file_list:
+            target_folder = (
+                functions.get_folder_from_s3_path(s3_key)
+                .replace(f"{enums.BucketPath.STAGING.value}/", f"{enums.BucketPath.UPLOAD.value}/")
+                .replace(f"/{metadata.study}/", f"/{metadata.study}/{file.split('.')[0]}/")
+            )
+            s3_client.upload_fileobj(
+                archive.open(file), Bucket=s3_bucket_name, Key=f"{target_folder}/{file}"
+            )
+    archive_key = s3_key.replace(
+        f"{enums.BucketPath.STAGING.value}/", f"{enums.BucketPath.ARCHIVE.value}/"
+    )
+    functions.move_s3_file(
```

Comment:
nit: maybe worth archiving this first and then reading from the archived version? Just so that there isn't an (admittedly unlikely) race condition between the manifest being written and this source file being overwritten by the next upload (when locking enforces a queue)

Reply:
But isn't it also nice to be able to debug from a canonical location, and not a location that could have been overridden since the event happened? Like, it makes sense to my brain to prioritize getting the "incoming landing zone, will be overriden" path into a "canonical, never-changing" path for processing at leisure.

But this is not a big deal and if you think locking solves this and prefer debugging without moving early, then let's keep it as it is.

Reply:
Compromise position: I'm going to leave the way this is now, but created #195 to cover a better solution for detecting errors, which would obviate my initial concern and make this easier to reason over.